pyproject.toml

```diff
@@ -54,6 +54,53 @@ dependencies    = [
     "yarl==1.8.1",
 ]
 
+[project.optional-dependencies]
+test = [
```

Comment:
In the current state, I think this is not the right thing to do.

IMHO it makes no sense to create a duplicate of the existing requirements file. Additionally, why would test requirement be in the `pyproject` file and not the others?

Currently, we only keep core requirements here, which IMHO is good enough for now.

Considering this change had no motivation, or solving an known problem case, I think it is safe to revert this one.

Reply:
Your comment doesn't make sense to me. My comment is in this case is about these requirements not being kept in this file, just like we don't keep any other requirement for other integrations in this file.

My comment was not about tests, recorder, stream or anything like that. I think you mixed up context.

Reply:
I'm sorry – this does not move any integration requirements to pyproject.toml?

If you look at the commit doing this move in isolation https://github.com/home-assistant/core/pull/81795/commits/200276f9d7bf8d66abc43e877f0cd5d3ab57c230 you can see no new requirements that hadn't been there before are being written into the generated txt file.

Reply:
> I'm sorry – this does not move any integration requirements to pyproject.toml?

Correct, so test requirements don't belong here either.

Reply:
Why couldn't they belong here? Before this PR, `requirements_test.txt` is the only `requirements*.txt` file that is not generated by a script.

Reply:
> is the only requirements*.txt file that is not generated by a script.

So? I miss the reasoning behind that? It's not generated 👍 That is perfectly fine?

> Why couldn't they belong here?

They could, but we don't. We only keep our core requirements in there, all others are in their own requirements files. Adding test requirements but not the others, is, therefore odd and should just not be done.

Reply:
> So? I miss the reasoning behind that? It's not generated 👍 That is perfectly fine?

Because "There should be one – and preferably only one – obvious way to do it." – [The Zen of Python](https://peps.python.org/pep-0020/#the-zen-of-python)

... where "it" in this case would be "maintain requirements files in home-assistant/core". Right now `requirements-test.txt` is the only requirements*txt file maintained by hand, everything else is mercilessly overwritten by a computer. Why would `requirements-test.txt` be the odd one out, other than "we've done it this way so far"?

> They could, but we don't. We only keep our core requirements in there, all others are in their own requirements files. 

They are in their own requirements files because a script put them there, not because someone "keeps" them there. 

> Adding test requirements but not the others, is, therefore odd and should just not be done.

Having test requirements as an optional-dependencies table is [not odd and is certainly done (link to GitHub Code Search)](https://cs.github.com/?q=project.optional-dependencies%20language%3ATOML&scopeName=All%20repos&scope=).

Reply:
> Because "There should be one – and preferably only one – obvious way to do it." – [The Zen of Python](https://peps.python.org/pep-0020/#the-zen-of-python)

Let's keep that out of scope. I do have strong opinions on the Zen of Python that are not fit any public discussion and really out of bound here as well.

> Having test requirements as an optional-dependencies table is [not odd and is certainly done](https://cs.github.com/?q=project.optional-dependencies%20language%3ATOML&scopeName=All%20repos&scope=)

It is not odd in general, it is odd here as we don't keep other non-core dependencies in our `pyproject.toml`. (Which is the point you seem to be missing in from my comments).

Please revert this change, to not have these requirements in `pyprojec.toml`. Thanks 👍